### PR TITLE
Improve CI pipeline INTER-180

### PR DESCRIPTION
**Pull request by JuroUhlar**
* Added `Lint` and `Unit tests` jobs
* Added caching for [Next build cache](https://nextjs.org/docs/pages/building-your-application/deploying/ci-build-caching#github-actions), [node_modules](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#using-the-cache-action), and [Playwright browser binaries](https://github.com/microsoft/playwright/issues/7249), reducing the best possible E2E test Github action time (if caches are hit) from cca 4.5 minutes to 2.5 minutes 
* Considered using Playwright Docker container instead of installing Playwright dependencies, but the docker overhead endep taking more time, details in the task: https://fingerprintjs.atlassian.net/jira/software/c/projects/INTER/boards/13?assignee=712020%3A733fd86d-af57-4843-91a0-a559847037b3&selectedIssue=INTER-180
* Upgraded some dependencies, fixed some lint errors

### CODE REVIEW

**Review by TheUnderScorer — DISMISSED** [review]

File: playwright.config.js [hunk 1]
```diff
@@ -1,16 +1,17 @@
-const { defineConfig, devices } = require('@playwright/test');
-
+import { defineConfig, devices } from '@playwright/test';
 /**
  * Read environment variables from file.
- * https://github.com/motdotla/dotenv
+ * https://github.com/motdotla/dotenv;
```
Comment (ilfa) [inline playwright.config.js:4]: Do you really need this semicolon?

File: src/client/components/common/content.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/jsx-key */
```
Comment (ilfa) [inline src/client/components/common/content.tsx:1]: Why not just add it? =)

**Review by ilfa — APPROVED** [review]

File: playwright.config.js (hunk 1 above)
Comment (JuroUhlar) [reply playwright.config.js]: Good catch, fixed!

File: src/client/components/common/content.tsx (hunk 2 above)
Comment (JuroUhlar) [reply src/client/components/common/content.tsx]: You are right, fixed!

**Review by TheUnderScorer — APPROVED** [review]

**Review by ilfa — APPROVED** [review]

**Pull request merged**